pkg/client/e2e.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2018 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/heptio/sonobuoy/pkg/results"
+	"github.com/pkg/errors"
+)
+
+// GetTests extracts the junit results from a sonobuoy archive and returns the requested tests.
+func (c *SonobuoyClient) GetTests(reader io.Reader, show string) (TestCases, error) {
+	read := results.NewReaderWithVersion(reader, "irrelevant")
+	junitResults := jUnitResults{}
+	err := read.WalkFiles(func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		return results.ExtractFileIntoStruct("plugins/e2e/results/junit_01.xml", path, info, &junitResults)
```

Comment:
I think we may want to add a //TODO here about extensibility for things like the extended tests/sonobuoy-plugins

Reply:
I'll open a ticket to capture the idea of extending this command to work with our extended tests.

Reply:
https://github.com/heptio/sonobuoy/issues/306